test/string_extension_test.dart

```diff
@@ -45,4 +45,49 @@ void main() {
 
     expect(result, isEmpty);
   });
+
+  test('unMarkdownLinks should replace links when there is only one link', () {
+    const messageWithOneLinkUnMarkdowned = "https://superman.com/this-is-a-*superman*-link";
+    const messageWithOneLinkMarkdowned = "https://superman.com/this-is-a-<em>superman</em>-link";
+
+    final result = messageWithOneLinkMarkdowned.unMarkdownLinks(messageWithOneLinkUnMarkdowned);
+
+    expect(result, equals(messageWithOneLinkUnMarkdowned));
+  });
+
+  test('unMarkdownLinks should replace links when there are multiple links', () {
+    const messageWithMultipleLinksUnMarkdowned = "https://superman.com/this-is-a-*superman*-link https://batman.com/this-is-a-*batman*-link";
+    const messageWithMultipleLinksMarkdowned = "https://superman.com/this-is-a-<em>superman</em>-link https://batman.com/this-is-a-<em>batman</em>-link";
+
+    final result = messageWithMultipleLinksMarkdowned.unMarkdownLinks(messageWithMultipleLinksUnMarkdowned);
+
+    expect(result, equals(messageWithMultipleLinksUnMarkdowned));
+  });
+
+  test('unMarkdownLinks should replace links when there are links and text', () {
```

Comment:
how about the case:
1. `is this serious https://superman.com/this-is-a-<em>superman</em>-link`
2. `is this serious-https://superman.com/this-is-a-<em>superman</em>-link`
3. `https://superman.com/this-is-a-<em>superman</em>-link hello guys`
4. `https://superman.com/this-is-a-<em>supe hello guys`
5. `https://superman.com/this-is-a-<em>superman</em>-link hello guys https://bat`
6. `https://superman.com/this-is-a-<em>superman</em>-link hello guys https://batman.com/this-`

Reply:
for me 3,4,5,6 are already covered
I'll put a test for 1 for text + link and 2 to test the regex https

Reply:
i can not find 3,4,5,6 in test-cases. please separate it in method. Thanks

Reply:
updated